odl/tomo/backends/astra_cuda.py

```diff
@@ -228,37 +229,72 @@ def astra_cuda_back_projector(proj_data, geometry, reco_space, out=None):
 
     # Angular integration weighting factor
     # angle interval weight by approximate cell volume
-    extent = float(geometry.motion_partition.extent())
-    size = float(geometry.motion_partition.size)
-    scaling_factor = extent / size
+    angle_extent = float(geometry.motion_partition.extent())
+    num_angles = float(geometry.motion_partition.size)
+    scaling_factor = angle_extent / num_angles
+
+    astra_ver_num = astra.astra.version()
+    astra_version = '.'.join([str(astra_ver_num // 100),
+                              str(astra_ver_num % 100)])
+    if parse_version(astra_version) < parse_version('1.7'):
+        raise RuntimeError('ASTRA version < 1.7 not supported')
+    elif parse_version(astra_version) < parse_version('1.8'):
+        # Fix inconsistent scaling
+        if isinstance(geometry, Parallel2dGeometry):
+            # Scales with 1 / cell_volume
+            scaling_factor *= float(reco_space.cell_volume)
+        elif isinstance(geometry, FanFlatGeometry):
+            # Scales with 1 / cell_volume
+            scaling_factor *= float(reco_space.cell_volume)
+            # Additional magnification correction
+            src_radius = geometry.src_radius
+            det_radius = geometry.det_radius
+            scaling_factor *= ((src_radius + det_radius) / src_radius)
+        elif isinstance(geometry, Parallel3dAxisGeometry):
+            # Scales with voxel stride
+            # In 1.7, only cubic voxels are supported
+            voxel_stride = reco_space.cell_sides[0]
+            scaling_factor /= float(voxel_stride)
+        elif isinstance(geometry, HelicalConeFlatGeometry):
+            # Scales with 1 / cell_volume
+            # In 1.7, only cubic voxels are supported
+            voxel_stride = reco_space.cell_sides[0]
+            scaling_factor /= float(voxel_stride)
+            # Magnification correction
+            src_radius = geometry.src_radius
+            det_radius = geometry.det_radius
+            scaling_factor *= ((src_radius + det_radius) / src_radius) ** 2
+    else:
+        # TODO: check 2D
+        if isinstance(geometry, Parallel2dGeometry):
+            # Scales with 1 / cell_volume
+            scaling_factor *= float(reco_space.cell_volume)
+        elif isinstance(geometry, FanFlatGeometry):
+            # Scales with 1 / cell_volume
+            scaling_factor *= float(reco_space.cell_volume)
+            # Magnification correction
+            src_radius = geometry.src_radius
+            det_radius = geometry.det_radius
+            scaling_factor *= ((src_radius + det_radius) / src_radius)
+        elif isinstance(geometry, Parallel3dAxisGeometry):
+            # Scales with cell volume
+            # currently only square voxels are supported
+            scaling_factor /= reco_space.cell_volume
+        elif isinstance(geometry, CircularConeFlatGeometry):
```

Comment:
`CircularConeFlatGeometry` is a special case of `HelicalConeFlatGeometry`, why would be need two versions of the code?

The only difference is that `CircularConeFlatGeometry` has `pitch=0`.